lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp

```diff
@@ -46,10 +46,16 @@
 #include "lazy_tensor_core/csrc/ts_backend/ts_computation_client.h"
 #include "lazy_tensor_core/csrc/ts_backend/ts_lowering_context.h"
 #include "lazy_tensors/permutation_util.h"
+#include "lazy_tensor_core/csrc/ts_backend/LazyLazyIr.h"
 
 namespace torch_lazy_tensors {
 namespace compiler {
 
+// Forward decl of method that is codegenned and has no header
+TSOpVector LowerToTSCodegen(std::shared_ptr<torch::jit::GraphFunction> function,
```

Comment:
Not sure how this is linked to LazyTsLowering.cpp.

Reply:
This is the generated function that has the switch statement that calls into the specific lowering function per IR class.

Reply:
I understand that part. Just not quite exactly sure how this file is linked (linker) to that particular implementation.

Reply:
oh. nothing special, it's just how the linker works, I think! doing this forward decl is exactly the same as putting this line into a header file and including the header file here.  it causes the linker to declare an undefined `LowerToTSCodegen` in this compilation unit, and then later when the generated code .o is linked together with ts_node_lowerin.o into the same .so lib, this becomes resolved.

Reply:
Sure, if there is only one definition. What happens if there are two?

Reply:
I don't think there is any difference.  You can still mistakenly define foo in 2 compilation units, even if you use a header.  Even without the header, if you define it twice in the same link unit it'll complain in the same way as it complains if there were a header.  (Multiple definition error)

Reply:
This intrigues me because I have never done it or seem people forward declare a function without a header. And therefore wonder what could be the possible downsides.

Reply:
oh- to be clear, I'm not defending that a forward decl is a nice practice for our codebase.  I just did it out of laziness.  I think there's a >50% chance we completely redo the way the lowering code is architected.  One option is move all the lowering funcs as class methods on the node.  So, I didn't want to waste time making a new header file, although, that is silly because it is very easy to make a new file too.